Fix allow shortcuts to be updated when active profile changes

SlySven opened the pull request:
Without the uses of the `delete`s it is possible to get multiple `QShortcut`s defined and connected to various `mudlet` class slot methods which will then stop them working and trigger (at least in the KDE Plasma Desktop Environment) warnings about "Ambiguous keyboard shortcuts".

Also add a missing destructor to the `ShortcutsManager` class, which is desirable as it otherwise will not clean up some memory that will have been previously allocated with `new`.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
BugFix: allow shortcuts to be updated when active profile changes

Without the uses of the `delete`s it is possible to get multiple
`QShortcut`s defined and connected to various `mudlet` class slot methods
which will then stop them working and trigger (at least in the KDE Plasma
Desktop Environment) warnings about "Ambiguous keyboard shortcuts".

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Fix: add missing destructor to the ShortcutsManager class

This is desirable as it otherwise will not clean up some memory that will
have been previously allocated with `new`.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

Delwing commented on src/ShortcutsManager.cpp:
```diff
@@ -19,6 +20,24 @@
 
 #include "ShortcutsManager.h"
 
+ShortcutsManager::~ShortcutsManager()
```
Just a newbie question - so even if this class object is living for entire life of application it still needs destructor? Or is just good practice?

SlySven replied:
in this particular instance I suspect it is just *good practice* - always clean up after yourself...

SlySven replied:
BTW, why is the single instance of this class that is used created dynamically with `new` in the `mudlet` constructor? It is simpler to have a instance as a member of that class:
I.e. in the `mudlet.h` header you could change:
```diff
-    QPointer<ShortcutsManager> mShortcutsManager;
+    ShortcutsManager mShortcutsManager
```
Then in the constructor you do not need to create it (or explicitly destroy it - which isn't happening either at the moment):
```cpp
-    mShortcutsManager = new ShortcutsManager();
-    mShortcutsManager->registerShortcut(qsl("Script editor"), tr("Script editor"), &triggersKeySequence);
+    mShortcutsManager.registerShortcut(qsl("Script editor"), tr("Script editor"), &triggersKeySequence);
-    mShortcutsManager->registerShortcut(qsl("Show Map"), tr("Show Map"), &showMapKeySequence);
+    mShortcutsManager.registerShortcut(qsl("Show Map"), tr("Show Map"), &showMapKeySequence);
// etc...
```

As it is, the *current* name `mShortcutsManager` is a little confusing - it is actually a pointer (admittedly a *smart* one) and ***I*** (at least) like to have a `p` prefix for those as it helps to remember that outside of an IDE (like Qt Creator). If it had been my creation I would have called it `mpShortcutsManager` - of course if you follow the step I am hinting about above then the member would be the item concerned and NOT a pointer to it - so *then* it would be consistent to not have that `p` prefix. :grinning:

SlySven commented on src/ShortcutsManager.h:
```diff
@@ -27,11 +28,14 @@
 #include "post_guard.h"
 
-class ShortcutsManager : public QObject {
+class ShortcutsManager : public QObject
```
Ah, the **rule-of-three** strikes at me...

SlySven committed:
Revise: meet the rule-of-three

This might be enough to keep clang-tidy happy now that I've stuck in a
destructor...

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Fix: include deletion of the copy assignment operator

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Fix: include a move assignment operator

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Fix: add missing member to swap method needed by move constructor

Otherwise the constructor would be incomplete!

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven commented on src/ShortcutsManager.h:
```diff
@@ -27,18 +28,34 @@
 #include "post_guard.h"
 
-class ShortcutsManager : public QObject {
+class ShortcutsManager : public QObject
```
Aaaa5gh... :scream: It is dragging all five things out - even though most of them are, I think, unneeded!

SlySven committed:
Fix: delete all but default initialisation &  destructor special members

The others just are not needed.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>
